### PR TITLE
fixing issue with Youtube field which is always required

### LINKED ISSUE #2472: Youtube field is always required
When trying to post an content item from the admin side that has a youtube field it always fails form validation because the field is required if it's left empty. This field should be required only when specified ; not required all the time.

### PR DESCRIPTION
**Pull request by Skrypt**
Fixes #2472

### CODE REVIEW

**Comment by sebastienros** [comment]
It's missing a check in the driver too.

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs [hunk 1]
```diff
@@ -41,36 +47,38 @@ public override async Task<IDisplayResult> UpdateAsync(YoutubeField field, IUpda
...
+                var settings = context.PartFieldDefinition.Settings.ToObject<YoutubeFieldSettings>();
+                if (settings.Required && String.IsNullOrWhiteSpace(model.RawAddress))
                 {
-                    model.EmbeddedAddress = $"{uri.GetLeftPart(UriPartial.Authority)}/embed/{query["v"]}";
+                    updater.ModelState.AddModelError(Prefix, T["A value is required for {0}.", context.PartFieldDefinition.DisplayName()]);
                 }
-                else
-                {
-                    updater.ModelState.AddModelError(Prefix + "." + nameof(model.RawAddress), "The url format is not correct");
+                else {
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs]:
```suggestion
                else 
                {
```

**Comment by Skrypt** [comment]
Added the validation in driver

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs [hunk 2]
```diff
@@ -41,36 +47,38 @@ public override async Task<IDisplayResult> UpdateAsync(YoutubeField field, IUpda
...
-            if (!string.IsNullOrWhiteSpace(uri.Query))
-            {
-                var query = QueryHelpers.ParseQuery(uri.Query);
-                if (query.ContainsKey("v"))
+            if (await updater.TryUpdateModelAsync(model, Prefix)) {
+                var settings = context.PartFieldDefinition.Settings.ToObject<YoutubeFieldSettings>();
+                if (settings.Required && String.IsNullOrWhiteSpace(model.RawAddress))
                 {
-                    model.EmbeddedAddress = $"{uri.GetLeftPart(UriPartial.Authority)}/embed/{query["v"]}";
+                    updater.ModelState.AddModelError(Prefix, T["A value is required for {0}.", context.PartFieldDefinition.DisplayName()]);
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs]:
```suggestion
                    updater.ModelState.AddModelError(Prefix, T["A value is required for '{0}'.", context.PartFieldDefinition.DisplayName()]);
```

File: src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs [hunk 3]
```diff
@@ -41,36 +47,38 @@ public override async Task<IDisplayResult> UpdateAsync(YoutubeField field, IUpda
...
+                    if (!string.IsNullOrWhiteSpace(uri.Query))
+                    {
+                        var query = QueryHelpers.ParseQuery(uri.Query);
+                        if (query.ContainsKey("v"))
+                        {
+                            model.EmbeddedAddress = $"{uri.GetLeftPart(UriPartial.Authority)}/embed/{query["v"]}";
+                        }
+                        else
+                        {
+                            updater.ModelState.AddModelError(Prefix + "." + nameof(model.RawAddress), "The url format is not correct");
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.ContentFields/Drivers/YoutubeFieldDisplayDriver.cs]:
```suggestion
                            updater.ModelState.AddModelError(Prefix + "." + nameof(model.RawAddress), T["The format of the url is invalid"]);
```

**Comment by Skrypt** [comment]
Yeah, well I was doing it so it was faster. 😆

**Pull request merged**